community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/AggregationAcceptanceTest.scala

```diff
@@ -243,4 +243,19 @@ class AggregationAcceptanceTest extends ExecutionEngineFunSuite with NewPlannerT
 
     result.toList should equal(List(Map("foo" -> 42, "bar" -> 42, "baz" -> Map("y" -> 1))))
   }
+
+  test("should not project too much when aggregating in a WITH before merge and after a WITH with filter") {
```

Comment:
I don't understand what you are suggesting. If the bug will come back we don't know in which way it will manifest, we can hope that will fail again with an exception and hence this test will go red.